index.html

```diff
@@ -2276,22 +2284,133 @@ <h3 id="applying">
       </section>
       <section>
         <h3>
+          <code>purpose</code> member
+        </h3>
+        <p>
+          The <dfn><code>purpose member</code></dfn> is an <a>unordered set of
+          unique space-separated tokens</a> that are <a>ASCII
+          case-insensitive</a>. The allowed values the <a>icon purposes</a>.
+        </p>
+        <p>
+          When an <a>image object</a> is used as an <a>icon</a>, a developer
+          can specify that the image is intended to serve some special
+          <dfn>purpose</dfn> in the context of the host OS (i.e., for better
+          integration).
+        </p>
+        <p>
+          For example, as a <a data-lt="badge purpose">badge</a> or "pinned"
+          icon that is visually distinct, in color or form, from an
+          application's launch icon. The user agent uses the value of the
+          <a><code>purpose</code> member</a> as a hint to determine where and
+          how an <a>image object</a> is displayed. Unless declared otherwise by
+          the developer, a user agent can use an icon for <a>any purpose</a>.
+        </p>
+        <p>
+          The <dfn>icon purposes</dfn> are as follows:
+        </p>
+        <dl>
+          <dt>
+            <dfn data-lt="badge purpose">badge</dfn>:
+          </dt>
+          <dd>
+            A user agent can present this icon where space constraints and/or
+            color requirements differ from those of the application icon.
+          </dd>
+          <dt>
+            <dfn data-lt="any purpose">any</dfn>:
+          </dt>
+          <dd>
+            The user agent is free to display the icon in any context.
+          </dd>
+        </dl>
+        <p>
+          The <dfn>steps for processing a <code>purpose</code> member of an
+          image</dfn> are given by the following algorithm. The algorithm takes
+          an <a>image object</a> <var>image</var>. This algorithm will return a
+          set.
+        </p>
+        <ol>
+          <li>Let <var>purpose</var> be an empty set.
+          </li>
+          <li>Let <var>value</var> be the result of calling the
+          <a>[[\GetOwnProperty]]</a> internal method of <var>image</var>
+          passing " <code>purpose</code>" as the argument.
+          </li>
+          <li>Let <var>type</var> be <a>Type</a>(<var>value</var>).
+          </li>
+          <li>If <var>type</var> is not "string", then:
+            <ol>
+              <li>If <var>type</var> is not "<code>undefined</code>", <a>issue
+              a developer warning</a> that the type is unsupported.
+              </li>
+              <li>Return <code>undefined</code>.
+              </li>
+            </ol>
+          </li>
+          <li>Let <var>keywords</var> be the result of <a>splitting</a>
```

Comment:
I gave this more thought, and I think I want to stick with a string here - because it's rare that something will have multiple purpose (and we only define one purpose). Should probably only use arrays for complex types.